src/types.ts

```diff
@@ -17,63 +17,301 @@
 export type HttpMethod = 'post' | 'get';
 
 /**
- * HttpProtocol type
+ * DevicePlatform type
  */
-export type HttpProtocol = 'https' | 'http';
+export type DevicePlatform = 'web' | 'mob' | 'pc' | 'srv' | 'app' | 'tv' | 'cnsl' | 'iot';
 
 /**
- * The Subject Data
+ * LogLevel type
  */
-export type SubjectData = {
+export type LogLevel = 'off' | 'error' | 'debug' | 'verbose';
+
+/**
+ * BasisForProsessing
+ */
+export type Basis = 'consent' | 'contract' | 'legal_obligation' | 'legitimate_interests' | 'public_task' | 'vital_interests';
+
+/**
+ * BufferOption
+ */
+export type BufferOption = 'single' | 'default' | 'large';
+
+/**
+ * Size
+ */
+export type Size = [number, number];
```

Comment:
I'm commenting here since I'm reading through. I could figure out what this used for but `Size` is rather generic, can we think of a more descriptive name?